### ISSUE TITLE
Unifi: UDM-Pro 200% CPU usage from ctrld

### ISSUE THREAD

**Issue by GhostlyCrowd** [Control-D-Inc/ctrld#183]
I have a few routers I've been testing before I roll ctrld to my clients. Both of them seem to have 200% CPU usage from ctrld after a period of 24 hours, and stopping and then starting ctrld seems to resolve the issue (some times), however it has not lately and i have had to unsinstal it for the time being. 

![image](https://github.com/user-attachments/assets/c7ff07e3-1a3b-49df-aed8-56b1bb3a865e)

**Comment by yegors** [comment]
Would need to see the debug log at the time of this occurring. Can you run ctrld in logging mode, reproduce the issue, and post the log? If the log has sensitive info, email it to hello@controld.com and reference this issue.

**Comment by GhostlyCrowd** [comment]
> Would need to see the debug log at the time of this occurring. Can you run ctrld in logging mode, reproduce the issue, and post the log? If the log has sensitive info, email it to [hello@controld.com](mailto:hello@controld.com) and reference this issue.

I'm on dev now which is better but its still using a consistent 30% cCPU and a fair bit more memory than it was on 1.3.7 

I'll get you some logs, I'll try on my Personal crtrld first otherwise ill get some logs from the client I'm testing a deployment on on my msp account, Don't really want to soft lock their network again during office hours.

**Comment by GhostlyCrowd** [comment]
> Would need to see the debug log at the time of this occurring. Can you run ctrld in logging mode, reproduce the issue, and post the log? If the log has sensitive info, email it to [hello@controld.com](mailto:hello@controld.com) and reference this issue.

Sent the logs off to hello@

**Comment by yegors** [comment]
Have not received anything.

**Comment by GhostlyCrowd** [comment]
> Have not received anything.

Sent again.

**Comment by cuonglm** [comment]
@GhostlyCrowd Thanks for the information.

For confirmation, do you mean with site magic vpn running, the issue happens with v1.3.9, not v1.3.7?

**Comment by GhostlyCrowd** [comment]
> @GhostlyCrowd Thanks for the information.
> 
> For confirmation, do you mean with site magic vpn running, the issue happens with v1.3.9, not v1.3.7?

After further testing, I discovered there was also DNS looping on version 1.3.7. I've been using Site Magic for ages, but I recently started deploying ControlD Business and wanted to conduct some development testing on my office routing before rolling it out to clients.

My home UDM-Pro has my personal CTRLD account on it, and the two test endpoints had the MSP CTRLD installed. Unexpected issues arose ("DNS loops") because they were all linked via VPN Site Magic. All three routing systems started looping, and the CPU usage spiked to 200%. The development build of CTRLD 1.3.10 allowed them to function, although the CPU was still consistently +30% due to CTRLD 1.3.10, rather than the previous 200%.
